packages/components/dialog/src/dialog.ts

```diff
@@ -14,50 +18,89 @@ import styles from './dialog.scss.js';
  * @slot title - The title of the dialog
+ * @slot subtitle - The subtitle of the dialog
  */
 export class Dialog extends ScopedElementsMixin(LitElement) {
   /** @private */
   static get scopedElements(): ScopedElementsMap {
     return {
-      'sl-button-bar': ButtonBar
+      'sl-button': Button,
+      'sl-button-bar': ButtonBar,
+      'sl-icon': Icon
     };
   }
 
   /** @private */
-  static override styles: CSSResultGroup = styles;
+  static override styles: CSSResultGroup = [breakpoints, styles];
 
+  /** @private */
   @query('dialog') dialog?: HTMLDialogElement;
 
   /** Disables the ability to close the dialog using the Escape key. */
   @property({ type: Boolean, attribute: 'disable-close' }) disableClose = false;
 
+  /** Determines whether closing button (default one) should be shown in the top right corner. */
+  @property({ type: Boolean, attribute: 'closing-button' }) closingButton?: boolean;
+
   /** The ARIA role of the dialog. */
   @property() override role: 'dialog' | 'alertdialog' = 'dialog';
 
+  /** Alignment of the action buttons in the dialog footer
+   * @type {'start' | 'end'} */
+  @property({ reflect: true, attribute: 'buttons-align' }) buttonsAlign: Extract<ButtonBarAlign, 'start' | 'end'> =
+    'end';
+
   override connectedCallback(): void {
     super.connectedCallback();
 
     this.inert = true;
+
+    this.dialog?.setAttribute('closing', 'false');
+  }
+
+  override disconnectedCallback(): void {
+    this.dialog?.removeEventListener('animationend', event => this.#handleAnimationEnd(event));
+
+    super.disconnectedCallback();
   }
 
   override render(): TemplateResult {
     return html`
       <dialog
         @cancel=${this.#onCancel}
         @click=${this.#onClick}
+        @keydown=${this.#onKeydown}
         @close=${this.#onClose}
         .role=${this.role}
+        aria-labelledby="titleId"
         part="dialog"
       >
         <slot name="header">
-          <slot name="title"></slot>
+          <slot name="titles">
+            <slot name="subtitle"></slot>
+            <slot name="title" id="titleId"></slot>
+          </slot>
+          <slot name="header-actions">
+            <slot name="header-buttons"></slot>
+            ${this.closingButton
+              ? html`<slot name="close" @click=${this.#onCloseClick}>
+                  <sl-button fill="ghost" variant="default">
+                    <sl-icon name="xmark"></sl-icon>
+                  </sl-button>
+                </slot>`
+              : nothing}
+          </slot>
         </slot>
         <slot name="body">
-          <slot></slot>
+          <div class="body-wrapper">
+            <slot></slot>
+          </div>
         </slot>
         <slot name="footer">
-          <sl-button-bar align="end"><slot name="action"></slot></sl-button-bar>
+          <sl-button-bar class="footer-buttons" align=${this.buttonsAlign}><slot name="action"></slot></sl-button-bar>
```

Comment:
Do you need the class name here? Afaics there is no other `<sl-button-bar>` here? So the class name is not really necessary.

Reply:
Also `header-buttons` is a slot name. So having a class name called `footer-buttons` can be confusing.